src/os/bluestore/BlueStore.cc

```diff
@@ -670,7 +670,7 @@ void BlueStore::BufferSpace::read(
     Buffer *b = i->second.get();
     assert(b->end() > offset);
     if (b->is_writing() || b->is_clean()) {
-      if (b->offset < offset) {
+      if (b->offset <= offset) {
```

Comment:
this was deliberate.. the final if/else below handles the b->offset == offset case